cumulus/test/service/src/lib.rs

```diff
@@ -427,59 +424,61 @@ where
 	})?;
 
 	if let Some(collator_key) = collator_key {
-		let parachain_consensus: Box<dyn ParachainConsensus<Block>> = match consensus {
-			Consensus::RelayChain => {
-				let proposer_factory = sc_basic_authorship::ProposerFactory::with_proof_recording(
-					task_manager.spawn_handle(),
-					client.clone(),
-					transaction_pool.clone(),
-					prometheus_registry.as_ref(),
-					None,
-				);
-				let relay_chain_interface2 = relay_chain_interface_for_closure.clone();
-				Box::new(cumulus_client_consensus_relay_chain::RelayChainConsensus::new(
-					para_id,
-					proposer_factory,
-					move |_, (relay_parent, validation_data)| {
-						let relay_chain_interface = relay_chain_interface_for_closure.clone();
-						async move {
-							let parachain_inherent =
-							cumulus_client_parachain_inherent::ParachainInherentDataProvider::create_at(
-								relay_parent,
-								&relay_chain_interface,
-								&validation_data,
-								para_id,
-							).await;
-
-							let time = sp_timestamp::InherentDataProvider::from_system_time();
-
-							let parachain_inherent = parachain_inherent.ok_or_else(|| {
-								Box::<dyn std::error::Error + Send + Sync>::from(String::from(
-									"error",
-								))
-							})?;
-							Ok((time, parachain_inherent))
-						}
-					},
-					block_import,
-					relay_chain_interface2,
-				))
-			},
-			Consensus::Null => Box::new(NullConsensus),
-		};
-
-		#[allow(deprecated)]
-		old_consensus::start_collator(old_consensus::StartCollatorParams {
-			block_status: client.clone(),
-			announce_block,
-			runtime_api: client.clone(),
-			spawner: task_manager.spawn_handle(),
-			para_id,
-			parachain_consensus,
-			key: collator_key,
-			overseer_handle,
-		})
-		.await;
+		if let Consensus::Null = consensus {
```

Comment:
Not really related to this PR: why do we need Null consensus? When (and why) is it used?

Reply:
We use it in the [pov-recovery zombienet](https://github.com/paritytech/polkadot-sdk/blob/master/cumulus/zombienet/tests/0002-pov_recovery.toml#L42) test to have a authority that is not producing blocks.  Triggers different codepaths with reduced recovery delay.